darshan-util/pydarshan/tests/test_heatmap_handling.py

```diff
@@ -0,0 +1,561 @@
+                        50221.84291188226,
+                        540019.7821112294,
+                        418.49887595643366,
+                        108234.18114341467,
+                        942483.3199806289,
+                    ],
+                    [
+                        0,
+                        567426.7192208751,
+                        50754.88456134755,
+                        50754.88456134756,
+                        50754.88456134756,
+                        50754.884561347535,
+                        141846.72456250372,
+                        136490.77078712088,
+                        1050536.24718411,
+                        0,
+                    ],
+                    [
+                        0,
+                        1048576,
+                        0,
+                        0,
+                        0,
+                        0,
+                        0,
+                        251.0322619047617,
+                        1050548.9677380952,
+                        0,
+                    ],
+                ]
+            ),
+        ),
+    ],
+)
+def test_get_heatmap_data(
+    filepath,
+    expected_hmap_data,
+    xbins,
+    ops,
+):
+    # regression test for `heatmap_handling.get_heatmap_data()`
+
+    # generate the report and use it to obtain the aggregated data
+    report = darshan.DarshanReport(filepath)
+    agg_df = heatmap_handling.get_aggregate_data(
+        report=report, mods=["DXT_POSIX"], ops=ops
+    )
+    # run the aggregated data through the heatmap data code
+    actual_hmap_data = heatmap_handling.get_heatmap_data(agg_df=agg_df, xbins=xbins)
+
+    if filepath == "tests/input/sample-dxt-simple.darshan":
+        # check the data is conserved
+        assert actual_hmap_data.sum() == 4040
+        # make sure the output array is the correct shape
+        assert actual_hmap_data.shape == (1, xbins)
+        # make sure the output data contains identical values
+        assert_array_equal(actual_hmap_data, expected_hmap_data)
+
+    elif filepath == "examples/example-logs/dxt.darshan":
+        # make sure the output array is the correct shape
+        assert actual_hmap_data.shape == (1, xbins)
+        # make sure the output data contains identical values
+        assert_array_equal(actual_hmap_data, expected_hmap_data)
+
+        # for each combination of operations, make sure the sum is correct
+        if len(ops) == 2:
+            assert actual_hmap_data.sum() == 35539507
+        elif ops[0] == "read":
+            assert actual_hmap_data.sum() == 22517726
+        elif ops[0] == "write":
+            assert actual_hmap_data.sum() == 13021781
+
+    elif filepath == "examples/example-logs/ior_hdf5_example.darshan":
+        # make sure the output array is the correct shape
+        assert actual_hmap_data.shape == (4, xbins)
+        # make sure the output data contains identical values
+        assert np.allclose(actual_hmap_data, expected_hmap_data)
```

Comment:
It is far more helpful to use `assert_allclose()` here because it will actually report the extent of the mismatch/details, rather than using a plain assert on the plain NumPy function boolean return value.

Reply:
I replaced all instances of `assert np.allclose()` with `assert_allclose()`, so this can be resolved.